src/rc/start-stop-daemon.c

```diff
@@ -1005,7 +1024,15 @@ int main(int argc, char **argv)
 
 		for (i = getdtablesize() - 1; i >= 3; --i)
 			close(i);
-
+#ifdef HAVE_SCHED
+		if (scheduler != -1){
+			sched = malloc(sizeof(sched));
+			sched->sched_priority = sched_prio;
```

Comment:
you only initialize this field which means the rest are random garbage.  when declaring it on the stack, use a struct designated initializer to fix that.